File: iotcloud-restapi-php/Core/src/IoTCloud/Request/ListDevicesRequest.php
<?php

namespace TXIoTCloud\Request;

require_once "ListProductsRequest.php";

class ListDevicesRequest extends ListProductsRequest
{

    /**
     * 需要查看设备列表的产品 ID
     */
    private $productID;

    /**
     * CreateProductRequest constructor.
     * @param $timestamp     当前unix时间戳
     * @param $nonce         随机正整数，与timestamp联合起来，用于防止重放攻击
     * @param $pageSize      分页的大小
     * @param $pageNum       请求的页数
     * @param $productID     产品ID
     */
    public function __construct($timestamp, $nonce, $pageSize, $pageNum, $productID)
    {
        parent::__construct($timestamp, $nonce, $pageSize, $pageNum);
        $this->productID = $productID;
    }

    /**
     * 获取需要查看设备列表的产品 ID
     */
    public function getProductID()
    {
        return $this->productID;
    }

    /**
     * 设置需要查看设备列表的产品 ID
     * @param $productID
     */
    public function setProductID($productID)
    {
        $this->productID = $productID;
    }

}

File: iotcloud-restapi-java/Core/src/com/qcloud/restapi/iotcore/request/ListDevicesRequest.java
package com.qcloud.restapi.iotcore.request;

public class ListDevicesRequest extends Request {

    /**
     * 构造ListDevicesRequest对象
     *
     * @param productID 从官网获得的productID
     */
    public ListDevicesRequest(String productID) {
        put("Action", "ListDevices");
        put("productID", productID);
        put("pageSize", "10");
        put("pageNum", "1");
    }

    /**
     * 构造ListDevicesRequest对象
     *
     * @param productID 从官网获得的productID
     * @param pageSize
     * @param pageNum
     */
    public ListDevicesRequest(String productID, int pageSize, int pageNum) {
        put("Action", "ListDevices");
        put("productID", productID);
        put("pageSize", String.valueOf(pageSize));
        put("pageNum", String.valueOf(pageNum));
    }
}
